### PR TITLE
Enable loading of osm data for single submission

### LINKED ISSUE #75: Expose SlickGrid table events to library consumers
Example use case - Triggering a request to load data on double click.

### PR DESCRIPTION
**Pull request by okal**
Closes #75, subsumes #76.

### CODE REVIEW

File: src/hatti/views/table.cljs [hunk 1]
```diff
@@ -95,7 +95,17 @@
        :rowHeight 24
        :enableTextSelectionOnCells true})
 
-(defn sg-init [data form is-filtered-dataview?]
+(defn bind-external-slick-grid-event-handlers
```
Comment (okal) [inline src/hatti/views/table.cljs]: Any ideas on a testing strategy for this, that doesn't involve introspection into the `grid`, @pld?

File: src/hatti/ona/post_process.cljs [hunk 2]
```diff
@@ -56,16 +56,23 @@
...
             osm-val->osm-data (fn [osm-val]
-                                (when osm-val
-                                  (osm-id->osm-data
-                                   (osm-val->osm-id osm-val))))
+                                ;; The OSM-val can be either a string or a
+                                ;; precomputed osm-data value. This condition
+                                ;; ensures only strings are parsed for OSM ids
+                                (if (string? osm-val)
+                                  (if-let [osm-submission-data (osm-data
+                                                     (osm-val->osm-id osm-val))]
```
Comment (pld) [inline src/hatti/ona/post_process.cljs]: Indentation

File: src/hatti/views/table.cljs [hunk 3]
```diff
@@ -95,7 +95,17 @@
...
+  [grid event-handlers]
+  (doall
+   (for [[handler-key handler-function] event-handlers]
+     (let [handler-name (hyphen->camel-case (name handler-key))
```
Comment (pld) [inline src/hatti/views/table.cljs]: `for` has a `:let` key to simplify this syntax

File: src/hatti/views/table.cljs (hunk 1 above)
Comment (pld) [inline src/hatti/views/table.cljs]: Unfortunately no, could test at a higher level by adding a handler, initiating the event the handler is attached to, and checking that the handler is triggered as expect

**Pull request merged**